### DISCUSSION TITLE
common `loader1:loader0(xxx):file_name.suffix`

### DISCUSSION THREAD

**Discussion by xialvjun** [parcel-bundler/parcel#4913] [General]
we have `import a from 'url:a.png'`;
then why not `import b from 'text:b.txt'` and `import c from 'buffer:c.bin'`

Then we can `import gql_text from 'text:schema.gql'`. I know, I know, we have make the imported graphql to be just string rather than the ast.
But it's weired. If I have told parcel the suffix of the file is `.gql`, why I just get a string. If I just need a string, why to tell parcel the suffix of the file.

We can:
```js
import 'style:css:less:./style.less';
import str from 'text:./schema.gql';  // default utf8
import str from 'text("utf16"):./schema.gql';
import buffer from 'bin:./xxx.png';  // base64 in module, and get Buffer
import buffer from 'bin("base58"):./xxx.png';
import buffer from 'bin:./hello.wasm';  // base64 in module, and get Buffer
import hello from './hello.wasm';  // hello is a function
import url from 'url:./xxx.png';  // url = publicPath + __dirname + relativePath
```

**Comment by mischnic** [comment]
First of all, everything you are suggesting here except `text("utf16"):./schema.gql'` and `bin("base58"):./xxx.png ` (with the options in parentheses) can be used with the correct configuration and plugin. (And chained named pipelines `loader1:loaders2:./file.xyz` also don't work (yet?).)

If this is really about importing a GraphQL AST, you can still use a plugin that outputs the AST and specify in your `.parcelrc` that you want `gql:./schema.gql` to return the AST.

The current gql pipeline resolves imports, so it's not exactly the same as just importing the file's contents as a string.

**Reply by xialvjun** [reply]
In config file, it's global settings.
In `import text literal`, it's specific setting.
I believe easier specific setting is necessary.
We can have both global settings and specific setting at the same time.

**Comment by xialvjun** [comment]
```
// my.ts - a.ts will be bundled with my.ts
import a from './a.ts'; === import a from 'bundle:ts:./a.ts';

// my.ts - a.ts will not be bundled with my.ts, but a.ts will be translated to js and copied to dist
import a from 'url:ts:./a.ts'; === import a from 'ts:./a.ts';
// if a.ts has `import b from './b.ts'`, b.ts will be bundled with a.ts;

// if my.ts have common chunk as a.ts
// my.ts
import a from 'url:ts://a.ts';
import b from './b.ts';
// a.ts
import b from './b.ts';
// b.ts should not be a common chunk. There should be two b.ts, one is bundled with my.ts, one for a.ts.... 
// url is in fact act as entry
// lazy import().then  is different from two entry
```
